### PR TITLE
#27500 Prepare Developer module Tests for PHPUnit 8

### PR DESCRIPTION
**Pull request by lbajsarowicz**
### Description (*)
Preparing PHPUnit Tests to run with PHPUnit 8.
PRs for each module is separated to make tests easier to merge.

### Fixed Issues (if relevant)
1. magento/magento2#27500

### Manual testing scenarios (*)
N/A

### Questions or comments
PR #27701 needs to be merged first (otherwise PHPUnit tests fail)

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by lbajsarowicz** [comment]
@magento/community-engineering-team There is a very nasty trick used for the Unit test:
https://github.com/magento/magento2/blob/d4c156a51ab808dfed6ee42c6c4a22bc11b7d0da/app/code/Magento/Developer/Test/Unit/Console/Command/DevTestsRunCommandTest.php#L9-L26

I'm requesting manual approval. Later on, we can try to fix this.

**Comment by ihor-sviziev** [comment]
@lbajsarowicz I think static test failure might be related to incorrect namespace in phpunit file. Please try to fix ti

**Review by hws47a — COMMENTED** [review]
Changes look good to me. Let's see if we can make all CI tests pass

**Review by swnsma — CHANGES_REQUESTED** [review]
Hi @lbajsarowicz!
I want to say "thank you" for the work which you are doing.

Generally, PR looks very good.
If you would not like to fix a problematic file right now, could you add it to the [blacklist](https://github.com/magento/magento2/blob/2.4-develop/dev/tests/static/testsuite/Magento/Test/Php/_files/phpstan/blacklist/common.txt)?
We will have green builds and everyone will be happy.

**Comment by lbajsarowicz** [comment]
I have no idea how to perform that @swnsma

**Comment by swnsma** [comment]
@lbajsarowicz, please add file to: https://github.com/magento/magento2/blob/2.4-develop/dev/tests/static/testsuite/Magento/Test/Php/_files/phpstan/blacklist/common.txt

**Comment by lbajsarowicz** [comment]
@swnsma Done

**Comment by lbajsarowicz** [comment]
Blah, not this file, sorry. 
Let me change it.

File: dev/tests/static/testsuite/Magento/Test/Php/_files/phpcpd/blacklist/common.txt [hunk 1]
```diff
@@ -218,3 +218,4 @@ Magento/InventoryAdminUi/Controller/Adminhtml
 Magento/Newsletter/Model/Queue
 Magento/Framework/Mail/Template
 Magento/CheckoutAgreements/Model/Checkout/Plugin
+Magento/Developer/Test/Unit/Console/Command/DevTestsRunCommandTest.php
```
Comment (swnsma) [inline dev/tests/static/testsuite/Magento/Test/Php/_files/phpcpd/blacklist/common.txt]: Are you sure we also need to blacklist the file for PHP CD ?

Comment (lbajsarowicz) [reply dev/tests/static/testsuite/Magento/Test/Php/_files/phpcpd/blacklist/common.txt]: No we don't, just forgot to rollback the file change. One sec.

**Review by swnsma — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @swnsma, thank you for the review.
[ENGCOM-7409](https://jira.corp.magento.com/browse/ENGCOM-7409) has been created to process this Pull Request

**Review by hws47a — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @hws47a, thank you for the review.
[ENGCOM-7409](https://jira.corp.magento.com/browse/ENGCOM-7409) has been created to process this Pull Request

**Pull request merged**